src/functions/Add-DscResourceModule.ps1

```diff
@@ -0,0 +1,54 @@
+function Add-DscResourceModule {
+  <#
+  .SYNOPSIS
+    Vendor PowerShell modules into the build folder either by path or by retrieving the modules from the Gallery
+
+  .DESCRIPTION
+    Given a DSC module name (and optional version) this will download and unpack the module and its dependencies
+    to the local machine. See Save-Module for additional information on what is happening.
+
+
+  .PARAMETER Name
+    Specifies the name of the module to save.
+
+  .PARAMETER Path
+    Specifies the location on the local computer to store the saved module.
+
+  .PARAMETER RequiredVersion
+    Specifies the exact version number of the module to save.
+    If left blank, will default to latest available.
+
+  .EXAMPLE
+    Add-DscResourceModule -TargetDir ./tmp -Name PowerShellGet -Version 2.2.3
+
+    This example will search the PowerShell gallery for version `2.2.3` of the PowerShellGet and,
+    if it finds it, save the module and its dependencies into a folder called `./tmp`.
+  #>
+  [CmdletBinding()]
+  param (
+    $Name,
+    $Path,
+    $RequiredVersion
+  )
+
+  Begin { }
+
+  Process {
+    if (-not(Test-Path $Path)) {
+      if (-not(Test-Path $Path)) {
+        New-Item -Path $Path -Force -ItemType 'Directory'
+      }
+      $PathTmp = $Path -Replace "(/|\\)$", '_tmp'
+      if (-not(Test-Path $PathTmp)) {
+        New-Item -Path $PathTmp -Force -ItemType 'Directory'
+      }
+      Save-Module -Name $Name -Path $PathTmp -RequiredVersion $RequiredVersion
+      ForEach ($ModuleFolder in (Get-ChildItem $PathTmp)) {
+        Move-Item -Path (Get-ChildItem $ModuleFolder.FullName).FullName -Destination "$Path/$($ModuleFolder.Name)"
```

Comment:
For some reason in the [acceptance testing](https://ci.appveyor.com/project/puppetlabs/puppetdscbuilder/builds/32196638#L82) it looks like the evaluation of `(Get-ChildItem $ModuleFolder.FullName).FullName` is matching `"$Path/$($ModuleFolder.Name)"`

On further investigation in debugging, it looks like this is because the solution to strip trailing backslashes actually only appends `_tmp` if the path ended in `\` or `/`.

It looks like the replace line should instead be:

```powershell
$PathTmp = ($Path -Replace "(/|\\)$", $Null) + '_tmp'
```

We should probably update the testing to verify what happens when a path is specified _without_ a trailing slash as well as with one.

Reply:
:facepalm: